src/main/java/tc/oc/pgm/PGMImpl.java

```diff
@@ -339,7 +340,10 @@ public Identity getIdentity(UUID playerId, String username, @Nullable String nic
     prefixRegistry = new PrefixRegistryImpl();
     registerEvents(prefixRegistry);
 
-    new RestartManager(this);
+    if (Config.AutoRestart.enabled()) {
+      new RestartManager();
```

Comment:
`RestartManager.setInstance(new RestartManager());`?

Reply:
The restart manager constructor does this anyway.

Reply:
Yes, but setting static variables from objects (non-static context) isn't right. If you don't want to call `setInstance` you can always use static initializer.

```java
private static final Kebabtallrik MEAL;

static {
    MEAL = new Kebabtallrik();
}
```